src/TLuaInterpreter.cpp

```diff
@@ -1556,14 +1556,19 @@ int TLuaInterpreter::getLineCount(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getColumnNumber
 int TLuaInterpreter::getColumnNumber(lua_State* L)
 {
+    QString windowName;
     Host& host = getHostFromLua(L);
     if (lua_isstring(L, 1)) {
-        string window = lua_tostring(L, 1);
-        QString _window = window.c_str();
-        lua_pushnumber(L, mudlet::self()->getColumnNumber(&host, _window));
+        if (!lua_isstring(L, 1)){
+            lua_pushfstring(L, "getColumnNumber: bad argument #1 type (window name as string expected, got %s!)", luaL_typename(L, 1));
+            return lua_error(L);
+        }
+        windowName = QString::fromUtf8(lua_tostring(L, 1));
+        lua_pushnumber(L, mudlet::self()->getColumnNumber(&host, windowName));
```

Comment:
Up to now it just returned `-1` if a window name was specified and that window did not exist - I think we can change that to returning the more normal lua paradigm of returning two items - a nil and then a error message as a string - and @druuimai has implemented it.

This does change the API - but ONLY for the case where a non-existent `TConsole` was requested - and this was the point I was trying to emphasise: in that this is (***I*** think) where we can break backward compatibility because it does not change things for the working cases - it just provides better feedback when things don't.

Reply:
Seems alright to me. @Mudlet/lua-interface ?